lib/permutations.rb

```diff
@@ -1,4 +1,27 @@
+# Time complexity: O(n) => Because it is not a nested loop and it loops through depending on the input size. It changes from O(2n) to O(n) after dropping a constant, 2.
+
+# Space complexity: O(n) => Because one hash was created. 
 
 def permutations?(string1, string2)
-  raise NotImplementedError, "permutations? not implemented"
-end
\ No newline at end of file
+  # edge case
+  return false if string1.length != string2.length
+
+  hash = {} 
+
+  i = 0
+  while i < string1.length
+    hash[string1[i]] = true 
+
+    i += 1
+  end  
```

Comment:
What if you had more than 1 instance of a letter

You would do better to count the number of each letter.